src/test/java/plugins/MatrixAuthPluginTest.java

```diff
@@ -48,14 +49,23 @@ public void helloWorld() throws Exception {
         jenkins.login().doLogin("alice");
 
         FreeStyleJob j = jenkins.jobs.create();
+
         j.save();
 
         // if we login as Bob, he shouldn't see the job
         jenkins.login().doLogin("bob");
+
+        //wait for main panel to appear to make sure page is rendered
+        waitFor(by.id("main-panel"));
+
         assertNull(getElement(by.href("job/"+j.name+"/")));
 
         // contorl assertion: alice shoudl see the link
         jenkins.login().doLogin("alice");
+
+        //wait for main panel to appear to make sure page is rendered
+        waitFor(by.id("main-panel"));
```

Comment:
@Mobrockers 

> And besides, let's face it, this is a UI test. UI's change, and when UI's change tests have to be fixed. Such is life.

Well, we have to try finding the most robust way.

I'm going to check it to provide an alternative.